pkg/engine/events.go

```diff
@@ -278,7 +285,11 @@ func queueEvents(events chan<- Event, buffer chan Event, done chan bool) {
 				if !ok {
 					// If the event source has been closed, flush the queue.
 					for _, e := range queue {
-						events <- e
+						if !emitter.isClosed() {
+							events <- e
```

Comment:
This isn't right but tricky. emitter.isClosed() is always true here. However if we don't guard here we have another panic in the repro. The events channel gets closed outside of the event emitter code.